lib/testing/src/lib/core/pages/card-view-items/card-view-text-item.page.ts

```diff
@@ -0,0 +1,77 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { by, element } from 'protractor';
+import { BrowserVisibility } from '../../utils/browser-visibility';
+import { BrowserActions } from '../../utils/browser-actions';
+
+export class CardViewTextItemPage {
+
+    toggleTextLocator = by.css('div[data-automation-id*="card-textitem-edit-toggle"]');
+    editInputLocator = by.css('input[data-automation-id*="card-textitem-editinput"]');
+    clearIconLocator = by.css('mat-icon[data-automation-id*="card-textitem-reset"]');
+    saveIconLocator = by.css('mat-icon[data-automation-id*="card-textitem-update"]');
+    textValueLocator = by.css('span[data-automation-id*="card-textitem-value"]');
+    errorMessageLocator = by.css(`mat-error[data-automation-id*='card-textitem-error'] li`);
+    editIcon = by.css('mat-icon[data-automation-id*="card-textitem-edit-icon"]');
+    item;
+
+    constructor(item) {
+        this.item = element(by.xpath(`//div[@data-automation-id="card-textitem-label-${item}"]/ancestor::adf-card-view-textitem`));
+    }
+
+    getLabelValue() {
+        return this.item.element(by.css('div[data-automation-id*="label"]')).getText();
```

Comment:
The getText methods be used from BrowserActions by passing the elementFinder in most of the places.